app/main.py

```diff
@@ -1,2 +1,131 @@
+
+from typing import List, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, elements: List[tuple] = None) -> None:
+        if elements is None:
+            elements = []
+        self._bucket_size = 8
+        self._bucket_resize = self._bucket_size * 2 / 3
+        while len(elements) > self._bucket_resize:
+            self._bucket_size *= 2
+            self._bucket_resize = self._bucket_size * 2 / 3
+        self._buckets = [[] for i in range(self._bucket_size)]
+        if len(elements):
+            self._assign_buckets(elements)
```

Comment:
Your init method looks a bit mess and it is hard to read. Why just don’t made all the checks into the methods below?

Reply:
During initialization, a check must be performed to create the table, and then everything else.

Or you can do it another way?

Reply:
@katevasilenko , menched that you can create some method to check and then call it into init to reduce code in init method